src/io/fabric8/Fabric8Commands.groovy

```diff
@@ -486,7 +486,7 @@ def waitUntilSuccessStatus(project, ref) {
 
     def apiUrl = new URL("https://api.github.com/repos/${project}/commits/${ref}/status")
     waitUntil {
-        def HttpURLConnection connection = apiUrl.openConnection()
+        HttpURLConnection connection = apiUrl.openConnection()
         if (githubToken != null && githubToken.length() > 0) {
```

Comment:
isn't the idiom `githubToken?.length() > 0 ` ?

Reply:
If the token itself can be null, you will need `githubToken && githubToken.length() > 0` at least. 

I'm sure we will find a lot more like this, no need to fix it all in one patch.